internal/reconciler/logpipeline/status.go

```diff
@@ -105,8 +105,8 @@ func (r *Reconciler) evaluateConfigGeneratedCondition(ctx context.Context, pipel
 		return metav1.ConditionFalse, conditions.ReasonUnsupportedLokiOutput, conditions.MessageForLogPipeline(conditions.ReasonUnsupportedLokiOutput)
 	}
 
-	if secretref.ReferencesNonExistentSecret(ctx, r.Client, pipeline) {
-		return metav1.ConditionFalse, conditions.ReasonReferencedSecretMissing, conditions.MessageForMetricPipeline(conditions.ReasonReferencedSecretMissing)
+	if err := secretref.VerifySecretReference(ctx, r.Client, pipeline); err != nil {
+		return metav1.ConditionFalse, conditions.ReasonReferencedSecretMissing, err.Error()
```

Comment:
we might want to check what type of error was returned. If the VerifySecretReference returns some internal client error we would also use it as the message and we should probably replace this by some generic message and put more details in the logs

Reply:
Currently only the well-defined errors can happen. I agree that we should re-think the approach here but again, I think that should not be the concern of this PR